src/Behavioral.Automation/Services/ComplexBindingBuilder.cs

```diff
@@ -0,0 +1,131 @@
+        private readonly ScenarioContext _scenarioContext;
+
+        private readonly int _indentationSize = 4;
+        private int _indendationLevel = 1;
+
+        public ComplexBindingBuilder(
+            ITestRunner runner,
+            IStepParametersProcessor stepParametersProcessor,
+            ScenarioContext scenarioContext)
+        {
+            _runner = runner;
+            _stepParametersProcessor = stepParametersProcessor;
+            _scenarioContext = scenarioContext;
+        }
+
+        public void Indent()
+        {
+            _indendationLevel++;
+        }
+
+        public void ReduceIndentation()
+        {
+            _indendationLevel--;
+        }
+
+        public void BuildAction(Action method)
+        {
+            IEnumerable<StepDefinitionBaseAttribute> attributes = method.GetMethodInfo().GetCustomAttributes<StepDefinitionBaseAttribute>(true);
+            BuildAction(Array.Empty<object>(), attributes, method.GetMethodInfo().Name);
+        }
+
+        public void BuildAction<T>(Action<T> method, params object[] pars)
+        {
+            IEnumerable<StepDefinitionBaseAttribute> attributes = method.GetMethodInfo().GetCustomAttributes<StepDefinitionBaseAttribute>(true);
+            BuildAction(pars, attributes, method.GetMethodInfo().Name);
+        }
+
+        public void BuildAction<T1, T2>(Action<T1, T2> method, params object[] pars)
+        {
+            IEnumerable<StepDefinitionBaseAttribute> attributes = method.GetMethodInfo().GetCustomAttributes<StepDefinitionBaseAttribute>(true);
+            BuildAction(pars, attributes, method.GetMethodInfo().Name);
+        }
+
+        public void BuildAction<T1, T2, T3>(Action<T1, T2, T3> method, params object[] pars)
+        {
+            IEnumerable<StepDefinitionBaseAttribute> attributes = method.GetMethodInfo().GetCustomAttributes<StepDefinitionBaseAttribute>(true);
+            BuildAction(pars, attributes, method.GetMethodInfo().Name);
+        }
+
+        public void BuildAction<T1, T2, T3, T4>(Action<T1, T2, T3, T4> method, params object[] pars)
+        {
+            IEnumerable<StepDefinitionBaseAttribute> attributes = method.GetMethodInfo().GetCustomAttributes<StepDefinitionBaseAttribute>(true);
+            BuildAction(pars, attributes, method.GetMethodInfo().Name);
+        }
+
+        private void BuildAction(object[] pars, IEnumerable<StepDefinitionBaseAttribute> attributes, string methodName)
+        {
+            string definitionType = _scenarioContext.StepContext.StepInfo.StepDefinitionType.ToString();
+            var attributeForCurrentStep = attributes.FirstOrDefault(a => a.GetType().Name.StartsWith(definitionType, StringComparison.OrdinalIgnoreCase));
+            if (attributeForCurrentStep == null)
+            {
+                NUnit.Framework.Assert.Inconclusive($"{definitionType} attribute is not provided for {methodName} binding method");
```

Comment:
If `methodName` is supposed to be a calling method name you can use this example: 
https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.callermembernameattribute?view=net-6.0#examples

Reply:
Someone forgot to remove this. Now it's just Assert.Inconclusive